03-react-components/package.json

```diff
@@ -0,0 +1,25 @@
+{
+  "name": "02-react-setup",
+  "version": "1.0.0",
+  "description": "Eine Hello-World-Applikation mit React",
+  "main": "index.js",
+  "scripts": {
+    "start": "webpack-dev-server",
+    "compile": "webpack"
+  },
+  "author": "David Losert",
+  "license": "MIT",
+  "devDependencies": {
+    "@babel/cli": "7.8.4",
+    "@babel/core": "7.9.6",
+    "@babel/preset-react": "7.9.4",
+    "babel-loader": "^8.1.0",
+    "webpack": "^4.43.0",
+    "webpack-cli": "^3.3.11",
+    "webpack-dev-server": "^3.11.0"
+  },
+  "dependencies": {
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1"
+  }
```

Comment:
Zwei Anmerkungen:

- Zum einen fände ich es gut, wenn die `dependencies` vor den `devDependencies` kämen, denn das sind ja die eigentlich wichtigen Abhängigkeiten.
- Zum zweiten würde ich generell das `^` streichen (das halte ich persönlich für extrem schlechten Stil, damit zu arbeiten).

Reply:
Hab es angepasst.

Das mit den Carets sehe ich ehrlich gesagt wie du (auch wenn das mit der `package-json.lock` nicht mehr ganz so wild ist wie früher).

Das passiert ja per default aber automatisch bei `npm install --save` - hast du eine Ahnung, ob und wie ich das konfigurieren kann?

Reply:
Ja: In Deinem Homeverzeichnis liegt eine `.npmrc`-Datei (oder Du kannst sie anlegen). Darin muss folgende Zeile stehen:

```
save-exact=true
```

Danach lässt npm beim Installieren das Caret weg 😊